Rubberduck.CodeAnalysis/Inspections/Concrete/EmptyMethodInspection.cs

```diff
@@ -0,0 +1,131 @@
+    /// Sub Foo()
+    ///     ' ...
+    /// End Sub
+    /// ]]>
+    /// </example>
+    /// <example hasResults="false">
+    /// <![CDATA[
+    /// Sub Foo()
+    ///     MsgBox "?"
+    /// End Sub
+    /// ]]>
+    /// </example>
+    [Experimental(nameof(ExperimentalNames.EmptyBlockInspections))]
+    internal class EmptyMethodInspection : InspectionBase// EmptyBlockInspectionBase
+    {
+        public EmptyMethodInspection(RubberduckParserState state)
+            : base(state) { }
+
+        protected override IEnumerable<IInspectionResult> DoGetInspectionResults()
+        {
+            // Exclude empty members in user interfaces, as long as all members of the interface are empty,
+            // since some VB users might use concrete user defined classes as interfaces,
+            // while RD marks them as interfaces all the same.
+
+            return UserDeclarations.OfType<ModuleBodyElementDeclaration>()
+                .Where(bodyElement => !BlockContainsExecutableStatements(bodyElement.Block))
+                .GroupBy(bodyElement => bodyElement.ComponentName)
+                // Exclude results from user interfaces
+                .Where(bodyElements => !State.DeclarationFinder.FindAllUserInterfaces()
+                                       // where all members of that interface contain no executables
+                                       .Where(interfaceModule => interfaceModule.ComponentName == bodyElements.Key
+                                                                && interfaceModule.Members.Count == bodyElements.Count())
+                                       .Any())
+                .SelectMany(bodyElements => bodyElements)
+                .Select(result => new DeclarationInspectionResult(this,
+                                                                  string.Format(InspectionResults.EmptyMethodInspection,
+                                                                                result.DeclarationType.ToFormatted(),
+                                                                                result.IdentifierName),
+                                                                  result));
+        }
+
+        private bool BlockContainsExecutableStatements(BlockContext block)
+        {
+            return block?.children != null && ContainsExecutableStatements(block.children);
+        }
+
+        private bool ContainsExecutableStatements(IList<Antlr4.Runtime.Tree.IParseTree> blockChildren)
+        {
+            foreach (var child in blockChildren)
+            {
+                if (child is BlockStmtContext blockStmt)
+                {
+                    var mainBlockStmt = blockStmt.mainBlockStmt();
+
+                    if (mainBlockStmt == null)
+                    {
+                        continue;   //We have a lone line lable, which is not executable.
+                    }
+
+                    System.Diagnostics.Debug.Assert(mainBlockStmt.ChildCount == 1);
```

Comment:
`Debug.Assert` calls don't compile into a release build; if this assert fails, the next line could throw an exception. If it doesn't fail... it should be removed - or better, turned into an explicit guard clause ;-)

Reply:
IDK actually, after looking deeply into it, I see no reason for this assertion to exist, as all requirements for it are necessarily met in all cases. It could never fail, so I'll remove it.

Reply:
FWIW there is merit to "documentary debug asserts". They allow you to specify precoditions in code without necessarily forcing exception handling on you